plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/internal/ResultSetMessages.properties

```
@@ -27,6 +30,28 @@ controls_resultset_grouping_clear = Clear grouping
 controls_resultset_grouping_default_sorting = Default sorting
 controls_resultset_grouping_show_duplicates_only = Show duplicates only (COUNT > 1)
 
+controls_resultset_virtual_edit_action = Edit...
+controls_resultset_virtual_add_virtual_foreign_key = Add virtual foreign key
+controls_resultset_virtual_define_virtual_unique_key = Define virtual unique key
+controls_resultset_virtual_clear_virtual_unique_key = Clear virtual unique key
+controls_resultset_virtual_add_virtual_column = Add virtual column
+controls_resultset_virtual_foreignkey_page_remove_confirmation_question = Are you sure you want to delete virtual foreign key ''{0}''?
+controls_resultset_virtual_foreignkey_page_remove_confirmation_title  = Delete virtual FK
+controls_resultset_virtual_foreignkey_page_remove = Remove
+controls_resultset_virtual_foreignkey_page_add = Add
+controls_resultset_virtual_foreignkey_page_ref_datasource = Ref Datasource
+controls_resultset_virtual_foreignkey_page_columns = Columns
+controls_resultset_virtual_foreignkey_page_ref_table = Ref Table
+controls_resultset_virtual_foreignkey_page_text = Virtual Foreign Keys
+controls_resultset_virtual_columns_page_text = Virtual Columns
+controls_resultset_virtual_dictionary_page_text = Dictionary
+controls_resultset_virtual_keys_info_label = Entity logical structure is defined on client-side.\nYou can define virtual unique/foreign keys even if physical database\ndoesn't have or doesn't support them. Also you can define how to view column values.
+controls_resultset_viewer_frequent_refresh_hint = Hint: frequent refresh may cause high server load
+controls_resultset_viewer_switch_autocommit_hint = Hint: switch to auto-commit to see external changes
+controls_resultset_viewer_action_show_selected_row_count = Show selected row count
+controls_resultset_viewer_action_show_selected_column_count = Show selected column count
+controls_resultset_viewer_action_show_selected_cell_count = Show selected cell count
+controls_resultset_viewer_action_open_value_editor_column_readonly =  Column {0} is read-only: {1}
```

It might benefit from quotes:

```suggestion
controls_resultset_viewer_action_open_value_editor_column_readonly =  Column ''{0}'' is read-only: {1}
```